src/WebJobs.Script/Description/FunctionInvokerBase.cs

```diff
@@ -28,21 +27,42 @@ internal FunctionInvokerBase(ScriptHost host, FunctionMetadata functionMetadata,
         {
             Host = host;
             Metadata = functionMetadata;
+            _metrics = host.ScriptConfig.HostConfig.GetService<IMetricsLogger>();
 
+            // Function file logging is only done conditionally
             traceWriterFactory = traceWriterFactory ?? new FunctionTraceWriterFactory(functionMetadata.Name, Host.ScriptConfig);
             TraceWriter traceWriter = traceWriterFactory.Create();
+            FileTraceWriter = traceWriter.Conditional(t => Host.FileLoggingEnabled && (!(t.Properties?.ContainsKey(PrimaryHostTracePropertyName) ?? false) || Host.IsPrimary));
 
-            // Function file logging is only done conditionally
-            TraceWriter = traceWriter.Conditional(t => Host.FileLoggingEnabled && (!(t.Properties?.ContainsKey(PrimaryHostTracePropertyName) ?? false) || Host.IsPrimary));
-            _metrics = host.ScriptConfig.HostConfig.GetService<IMetricsLogger>();
+            // The global trace writer used by the invoker will write all traces to both
+            // the host trace writer as well as our file trace writer
+            TraceWriter = host.TraceWriter != null ? 
```

Comment:
This change is important - since the new SystemTraceWriter is part of the composite host trace writer, this ensures that invoker logs can also be traced to system (e.g. compilation results, etc.)